### DISCUSSION TITLE
Allow setting subject in reset password request email

### DISCUSSION THREAD

**Discussion by kayhanb** [directus/directus#14046] [Feature Requests]
`services > users.ts`

```js
async requestPasswordReset(email: string, url: string | null, subject?: string | null): Promise<void>
```

service function getting subject parameter but javascript sdk and `controller > auth.ts > '/password/request'`  not accept subject parameter.

**Comment by rijkvanzanten** [comment]
I'm a little worried about the security implications of making the subject configurable from the outside. Anybody can request the password request email, so you could easily request the email for another user, and then spoof the subject line to something malicious 🤔

**Reply by kayhanb** [reply]
yes I agree with that, maybe solution can be add allow list env MAIL_SUBJECT_ALLOW_LIST  just like PASSWORD_RESET_URL_ALLOW_LIST

**Reply by henk23** [reply]
More options:
1) Make it a (translatable) setting in project settings.
2) Make it a translation string: $t:password_reset_mail_subject
3) Make it an explicit env variable (not an allow list): PASSWORD_RESET_MAIL_SUBJECT

It's funny acutally: Directus is so great in all kinds of ways. I've been looking for a CMS like this for a long time. It's so good in fact that the hardcoded password reset email subject is my biggest issue with this software. ;)

**Reply by rijkvanzanten** [reply]
@KayhanB Is the use case to update the subject that's used _everywhere_, or on a per-request basis?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
